### ISSUE TITLE
Support FILTER for GROUP_CONCAT

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#8543]
In the case of SQLite it looks like `GROUP_CONCAT` does support the `FILTER` clause. See examples given in documentation: https://www.sqlite.org/windowfunctions.html. Currently all `GroupConcat#filterWhere()` methods always throw an `UnsupportedOperationException`.

Further, `Function#SUPPORT_FILTER` should probably also list `SQLITE` as a supporting dialect. This should probably be covered by a separate issue.

**Comment by lukaseder** [comment]
It appears that SQLite has added support for the `FILTER` clause only in window functions. I vaguely recall having seen that feature when I added support for window functions in SQLite (#7865), and ignored it, because it is really an aggregate function feature, not a window function feature.

But in fact, we could indeed think about implementing native support for window function `FILTER` in SQLite, while continuing to emulate it for aggregate functions. I have created a separate issue for this: #8546

This issue here is really two issues:

1. Adding native support for `FILTER` in SQLite (see #8546)
2. Adding support for `FILTER` in `GROUP_CONCAT`, which also affects MySQL and other databases (and emulate it if needed)

I'll change this issue to be about `GROUP_CONCAT` only, and not SQLite specifically.
